mlflow/sklearn/__init__.py

```diff
@@ -465,6 +471,322 @@ def load_model(model_uri):
+        eval_dataset_id = id(eval_dataset)
+
+        run_id = self.get_run_id_for_model(model)
+        registered_dataset_list = self._eval_dataset_info_map[run_id][eval_dataset_name]
+
+        for i, id_i in enumerate(registered_dataset_list):
+            if eval_dataset_id == id_i:
+                index = i
+                break
+        else:
+            index = len(registered_dataset_list)
+
+        if index == len(registered_dataset_list):
+            # register new eval dataset
+            registered_dataset_list.append(eval_dataset_id)
+
+        return self.gen_name_with_index(eval_dataset_name, index)
+
+    def register_prediction_result(self, run_id, eval_dataset_name, predict_result):
+        """
+        Register the relationship
+         id(prediction_result) --> (eval_dataset_name, run_id)
+        into map `_pred_result_id_to_dataset_name_and_run_id`
+        """
+        value = (eval_dataset_name, run_id)
+        prediction_result_id = id(predict_result)
+        self._pred_result_id_to_dataset_name_and_run_id[prediction_result_id] = value
+
+        def clean_id(id_):
+            _get_autologging_metrics_manager()._pred_result_id_to_dataset_name_and_run_id \
+                .pop(id_, None)
+
+        # When the `predict_result` object being GCed, its ID may be reused, so register a finalizer
+        # to clear the ID from the dict for preventing wrong ID mapping.
+        weakref.finalize(predict_result, clean_id, prediction_result_id)
+
+    @staticmethod
+    def gen_metric_call_command(self_obj, metric_fn, *call_pos_args, **call_kwargs):
+        """
+        Generate metric function call command string like `metric_fn(arg1, arg2, ...)`
+        Note: this method include inspecting argument variable name.
+         So should be called directly from the "patched method", to ensure it capture
+         correct argument variable name.
+
+        :param self_obj: If the metric_fn is a method of an instance (e.g. `model.score`),
+           the `self_obj` represent the instance.
+        :param metric_fn: metric function.
+        :param call_pos_args: the positional arguments of the metric function call. If `metric_fn`
+          is instance method, then the `call_pos_args` should exclude the first `self` argument.
+        :param call_kwargs: the keyword arguments ofthe metric function call.
+        """
+
+        arg_list = []
+
+        def arg_to_str(arg):
+            if arg is None or np.isscalar(arg):
+                if isinstance(arg, str) and len(arg) > 32:
+                    # truncate too long string
+                    return repr(arg[:32] + '...')
+                return repr(arg)
+            else:
+                # dataset arguments or other non-scalar type argument
+                return _inspect_original_var_name(arg, fallback_name=f'<{arg.__class__.__name__}>')
+
+        param_sig = inspect.signature(metric_fn).parameters
+        pos_param_sig_keys = list(param_sig.keys())
```

Comment:
```suggestion
        arg_names = list(param_sig.keys())
```

Can we simplify the name of this variable?